nextroute/model_sanity_check.go

```diff
@@ -0,0 +1,163 @@
+package nextroute
+
+import (
+	"time"
+
+	"github.com/nextmv-io/sdk/connect"
+)
+
+// SanityCheckDepth is the depth of the sanity check.
+type SanityCheckDepth int
+
+const (
+	// SanityCheckLow checks if there is at least one move per plan unit.
+	SanityCheckLow SanityCheckDepth = iota
+	// SanityCheckMedium checks same as SanityCheckLow.
+	SanityCheckMedium
+	// SanityCheckHigh count for each plan unit if it fits on a vehicle and
+	// therefor how many units fit on a vehicle.
+	SanityCheckHigh
+)
+
+// ModelSanityCheck is the interface for the sanity check.
+type ModelSanityCheck interface {
+	// Check runs the sanity check. It returns the sanity check output and an
+	// error if the sanity check itself fails. Duration is the maximum time
+	// allowed for the sanity check to run. If the sanity check takes longer
+	// the remark will be "timeout". Depth is the depth of the sanity check.
+	Check(duration time.Duration, depth SanityCheckDepth) (SanityCheckOutput, error)
+}
+
+// SanityCheckOutput is the output of the sanity check.
+type SanityCheckOutput struct {
+	// Remark is the remark of the sanity check. It can be "ok", "timeout" or
+	// anything else that should explain itself.
+	Remark string `json:"remark"`
+	// Error is the error raised during the sanity check. It can be empty.
+	Error string `json:"error"`
```

Comment:
Can we clarify a bit more whether this would contain a summary error or is an internal error of the check operation that wouldn't be expected? Without having had a look at the details I would assume that this is a high-level error message about some sanity checks firing, as the internal check error is returned separately from the output.